api/src/main/java/pl/touk/widerest/api/orders/payments/PaymentController.java

```diff
@@ -0,0 +1,179 @@
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.extern.slf4j.Slf4j;
+import org.broadleafcommerce.common.payment.PaymentGatewayType;
+import org.broadleafcommerce.common.payment.dto.PaymentRequestDTO;
+import org.broadleafcommerce.common.payment.service.PaymentGatewayConfigurationService;
+import org.broadleafcommerce.common.payment.service.PaymentGatewayConfigurationServiceProvider;
+import org.broadleafcommerce.common.payment.service.PaymentGatewayCustomerService;
+import org.broadleafcommerce.common.payment.service.PaymentGatewayHostedService;
+import org.broadleafcommerce.common.payment.service.PaymentGatewayTransparentRedirectService;
+import org.broadleafcommerce.common.vendor.service.exception.PaymentException;
+import org.broadleafcommerce.core.catalog.domain.Category;
+import org.broadleafcommerce.core.order.domain.BundleOrderItem;
+import org.broadleafcommerce.core.order.domain.DiscreteOrderItem;
+import org.broadleafcommerce.core.order.domain.Order;
+import org.broadleafcommerce.core.order.domain.OrderItem;
+import org.broadleafcommerce.core.order.service.OrderService;
+import org.broadleafcommerce.core.order.service.type.OrderStatus;
+import org.broadleafcommerce.core.payment.service.OrderToPaymentRequestDTOService;
+import org.broadleafcommerce.profile.core.service.CustomerUserDetails;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import pl.touk.widerest.api.common.ResourceNotFoundException;
+import pl.touk.widerest.security.oauth2.ResourceServerConfig;
+import springfox.documentation.annotations.ApiIgnore;
+
+import javax.annotation.Resource;
+import javax.validation.Valid;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import static javaslang.API.$;
+import static javaslang.API.Case;
+import static javaslang.API.Match;
+import static javaslang.Predicates.instanceOf;
+
+@RestController
+@RequestMapping(value = ResourceServerConfig.API_PATH + "/orders/{orderId}/payment", produces = { MediaTypes.HAL_JSON_VALUE })
+@Api(value = "orders", description = "Order payments endpoint", produces = MediaTypes.HAL_JSON_VALUE)
+@Slf4j
+public class PaymentController {
+
+    @Resource(name = "blOrderService")
+    protected OrderService orderService;
```

Comment:
\[Checkstyle\] ERROR: Variable 'orderService' must be private and have accessor methods\.